### PR TITLE
add MSMetaData::getSpwIDs(), continuing refactor of MSFitsOutput

### PR DESCRIPTION
**Pull request by dmehring**
Ongoing refactoring of MSFitsOutput with help from MSMetaData. No MSFitsOutput functionality changes in this batch.

### CODE REVIEW

File: ms/MSOper/MSMetaData.cc [hunk 1]
```diff
@@ -1952,6 +1952,18 @@ vector<String> MSMetaData::getSpwNames() const {
 	return out;
 }
 
+std::set<uInt> MSMetaData::getSpwIDs() const {
+    Vector<Int> ddIDs = *_getDataDescIDs();
+    vector<uInt> ddIDToSpw = getDataDescIDToSpwMap();
```
Comment (juliantaylor) [inline ms/MSOper/MSMetaData.cc]: this is unused -> function returns ddis?

Comment (juliantaylor) [inline ms/MSOper/MSMetaData.cc]: also looks like it could all be const references instead of copies

**Comment by dmehring** [comment]
Good catch. That was a bug. I now use the appropriate mapping. Note that = should provide a reference to a casa::Vector() (and somewhat confusingly to me, explicitly using an & on that line to denote a reference causes the resulting Vector to be corrupted in one unit test, causing a segfault).

**Comment by juliantaylor** [comment]
ah right I missed that it was a capital V vector which is already a reference.

**Comment by juliantaylor** [comment]
not sure why it crashed as a reference though.

can the test be updated to catch that bug?

**Comment by dmehring** [comment]
I'm compiling with C++11 enabled, so not sure if that makes a difference or not.

Which bug? The reference bug or the bug I fixed? I really can't test the latter without swapping in a new dataset; in the current dataset there is only one polarization so the spwID is equal to the ddID which is why the test passed previously with the buggy version.

**Comment by juliantaylor** [comment]
hm ok, probably not worth adding more data to test such a small function, merging

**Pull request merged**